### ISSUE TITLE
Send automated emails after every action on the ticket

### ISSUE THREAD

**Issue by inghamn** [City-of-Bloomington/uReport#228]
We've added an auto response when the ticket is first opened.  However, we still require employees to manually email the reporter using their own email.  Employees would prefer not to use their own email accounts to respond to angry citizens.

Employees would prefer to have the system automatically send a response when the ticket is closed.  In addition, departments would like to provide a custom messages for the automated responses for each category.

We should look into providing automated responses for every action on a ticket.  The system should provide an interface for setting the custom message per category and action.

- [x] Move existing autoResponse on open ticket to new table
- [x] Interface for editing category_action_response templates
- [x] TicketHistory log entries need to check whether to send notifications
- [x] Figure out all the people for a ticket who should receive the response
- [x] Figure out which email address to use for each person
- [x] Create a syntax for inserting all the variables into the email message

**Comment by inghamn** [comment]
Do we want people to need to explicitly register to receive email notices?  Or do we just send out email notices to anyone and everyone involved in a ticket for whom we have an email address?

**Comment by rbdietz** [comment]
Probably worth discussing.  But no need to spam people with inconsequential
actions I would think. R

On Fri, May 27, 2016 at 12:15 PM, Cliff Ingham <notifications@github.com>
wrote:

> Do we want people to need to explicitly register to receive email notices?
> Or do we just send out email notices to anyone and everyone involved in a
> ticket for whom we have an email address?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/City-of-Bloomington/uReport/issues/228#issuecomment-222189091>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABFVevvw7qGnGTBsFQ6xwdE4uKFRK3B7ks5qFxiegaJpZM4IjQY0>
> .
>



-- 
~~~~~~~~~~~~~~~~~~~~~~~~~
Rick Dietz, Director
Information & Technology Services Dept.
City of Bloomington, Indiana
812.349.3485 (office) 812.340.7958 (cell)
dietzr@bloomington.in.gov

**Comment by hiesterd** [comment]
Does it make sense to send an initial email with a link to the report, so the reporter can check back for updates at whatever frequency they see fit?

**Comment by inghamn** [comment]
@hiesterd Yes, we already do that.  The email we send to the reporting person includes a URL to the ticket.

**Comment by rbdietz** [comment]
Yes, submitter should receive an email with the issue id and a link to view
it.

On Fri, May 27, 2016 at 1:11 PM, Dan Hiester <notifications@github.com>
wrote:

> Does it make sense to send an initial email with a link to the report, so
> the reporter can check back for updates at whatever frequency they see fit?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/City-of-Bloomington/uReport/issues/228#issuecomment-222201960>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABFVelAkC0fB922ncP7iFnR5yzM4HVP1ks5qFyWtgaJpZM4IjQY0>
> .
>



-- 
~~~~~~~~~~~~~~~~~~~~~~~~~
Rick Dietz, Director
Information & Technology Services Dept.
City of Bloomington, Indiana
812.349.3485 (office) 812.340.7958 (cell)
dietzr@bloomington.in.gov

**Comment by inghamn** [comment]
This fixes #109
